src/RageUtil.h

```diff
@@ -16,10 +16,24 @@
 
 class RageFileDriver;
 
-/** @brief Safely delete pointers. */
-#define SAFE_DELETE(p)       do { delete (p);     (p)=nullptr; } while( false )
-/** @brief Safely delete array pointers. */
-#define SAFE_DELETE_ARRAY(p) do { delete[] (p);   (p)=nullptr; } while( false )
+class RageUtil {
+public:
+	// Safely delete pointers.
+	template <typename T>
+	inline static void SafeDelete(T*& p) noexcept
```

Comment:
PR looks good but I'm curious why we're doing a reference to a pointer?

Reply:
so that uses of the original pointer are more likely to crash by dereferencing null instead of dereference a deleted pointer

I'd do away with this entirely and just `delete` without the helper method. you can always use asan builds to find these sorts of issues, nowadays there isn't as much reason to do this and the extra layer of abstraction is unnecessary IMO

Reply:
I agree it doesn't seem necessary, and probably more error prone than it needs to be. 

I just didn't want to change any functionality without having the PR checked out.

Reply:
I also agree with just deleting things in line instead of needing this abstraction, but if you prefer to merge this in first and then have a follow up that's okay too. Let me know which you prefer!

Reply:
> I also agree with just deleting things in line instead of needing this abstraction, but if you prefer to merge this in first and then have a follow up that's okay too. Let me know which you prefer!

Thanks, in that case please do merge this since I'd really like to get `RageUtil` in there so I can work off of it, and just in case I run into any unexpected dependencies on this behavior when I'm preparing the follow-up to get these deletions done inline.